### PR TITLE
DR 922 - Continuous Integration (TravisCI + SauceLabs)

### PR DESCRIPTION
**Pull request by gastonmancini**
### Backgroud

Continuous testing was incorporated into the CI process. Travis CI was integrated with Sauce Labs, a browser testing platform. It will allow us to have immediate feedback of the tests results, a historical timeline of metrics and will reduce integration overhead.

### Changes done

- Integrate SauceLabs and TravisCI
- Comment out 2 tests that are failing and require to be revisited (a new task on Jira was created for that DR-923)

### Note

A Confluence page was added where with the detailes of how we can generate the secure SauceLabs credentials. It was also added to the README.md file.

### Pending to be done

- Review why the bower components are not being cached by Travis
- Fix broken tests (DR-923)
- Impove README.md file with details about the project

### Useful links
* [Travis dashboard](https://travis-ci.org/DopplerRelay/relay-webapp)
* [SauceLabs dashboard](https://saucelabs.com/u/dopplerrelay)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  - sh -e /etc/init.d/xvfb start
+  - sudo apt-get update
+  - sudo apt-get install -y libappindicator1 fonts-liberation
+  - wget https://dl.google.com/linux/direct/google-chrome-stable_current_amd64.deb
+  - sudo dpkg -i google-chrome*.deb
+before_script:
+  # Install dependencies
+  - yarn
+  # Update the browser webdrivers
+  - node ./node_modules/protractor/bin/webdriver-manager update
```
Comment (andresmoschini) [inline .travis.yml]: Did you verify if it is not done by `yarn` when executes _postinstall_? see https://github.com/DopplerRelay/relay-webapp/blob/develop/src/package.json#L50

File: src/protractor-travis-conf.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  },
+  multiCapabilities: [{
+    'tunnel-identifier': process.env.TRAVIS_JOB_NUMBER,
+    build: process.env.TRAVIS_BUILD_NUMBER,
+    name: 'relay-webapp Chrome build ' + process.env.TRAVIS_BUILD_NUMBER,
+    browserName: 'chrome',
+    shardTestFiles: true,
+    maxInstances: 5,
+    seleniumVersion: '2.46.0'
+  }]
```
Comment (andresmoschini) [inline src/protractor-travis-conf.js]: I think that it is missing to call `onPrepare` script on this config, see https://github.com/DopplerRelay/relay-webapp/blob/develop/src/spec/config.js#L9

**Comment by andresmoschini** [comment]
Great work @gastonmancini !!! I know that we still have a lot of work to improve our process but it is a big step in the right direction. It was a wonderful thing to get you at least some days to work with us.

File: .travis.yml (hunk 1 above)
Comment (gastonmancini) [inline .travis.yml]: Removed. Work like a charm!

File: src/protractor-travis-conf.js (hunk 2 above)
Comment (gastonmancini) [inline src/protractor-travis-conf.js]: Done!

**Comment by andresmoschini** [comment]
❤️ :shipit:

**Pull request merged**